packages/Notification/Notification.jsx

```diff
@@ -109,15 +122,27 @@ class Notification extends React.Component {
   }
 
   render() {
-    if (
-      (this.props.dismissible || this.props.dismissibleA11yLabel) &&
-      !(this.props.dismissible && this.props.dismissibleA11yLabel)
-    ) {
+    const { dismissible, dismissibleA11yLabel, onExit, onDismiss } = this.props
+    const dismissibleHasA11yLabel = dismissible && dismissibleA11yLabel
+
+    if ((dismissible || dismissibleA11yLabel) && !dismissibleHasA11yLabel) {
       warn(
         'Notification',
         'The props `dismissible` and `dismissibleA11yLabel` must be used together.'
       )
     }
+    if (onExit && !dismissibleHasA11yLabel) {
+      warn(
+        'Notification',
+        'The props `onExit` must be used together with `dismissible` and `dismissibleA11yLabel`.'
```

Comment:
Typo, should be: The prop `onExit` must be used together with `dismissible` and `dismissibleA11yLabel`.